### DISCUSSION TITLE
question about initiating pyemu.utils.PstFrom object for an unstructured grid

### DISCUSSION THREAD

**Discussion by DataGeoranger** [pypest/pyemu#475] [Q&A] [answered]
Hello!

I am trying to initiate PstFrom object for a modflow-usg model, and one input it requires is the spatial reference object that offered by pyemu.helpers.SpatialReference.from_namfile(). 

However, there was a note in the code documentation that SpatialReference only works for structured grids. Has anyone set up the PstFrom object for an unstructured grid? Any ideas or thoughts are much appreciated!

**Comment by jtwhite79** [comment, marked as answer]
Hey @DataGeoranger  - we have a new tool help with unstructured grid parameterization and post processing called [pypestutils](https://github.com/pypest/pypestutils) and we have blazed a trail using it with pstfrom.  There is an example of using it with a quadtree mf6 model [here](https://github.com/pypest/pypestutils/blob/main/examples/pypestutils_pstfrom_quadtree_mf6.ipynb).  It has a pre-compiled Fortran lib that is staged for use with `pip` so just `pip install pypestutils` to get the installation locally.  If you are not using mf6, you can still use ppu, but might take a little more effort (ppu is generic for pilot points).  Hopefully that helps!

**Reply by DataGeoranger** [reply]
@jtwhite79 This is wonderful! Thank you for the tools, Jeremy!

**Reply by jtwhite79** [reply]
I cant take much credit for ppu - that was mostly @doherty1234 and @mwtoews

**Reply by DataGeoranger** [reply]
thanks all!
